src/main/rx/flysky.c

```diff
@@ -0,0 +1,496 @@
+/*
+* This file is part of Cleanflight.
+*
+* Cleanflight is free software: you can redistribute it and/or modify
+* it under the terms of the GNU General Public License as published by
+* the Free Software Foundation, either version 3 of the License, or
+* (at your option) any later version.
+*
+* Cleanflight is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+* GNU General Public License for more details.
+*
+* You should have received a copy of the GNU General Public License
+* along with Cleanflight.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+#include <string.h>
+#include "platform.h"
+
+#ifdef USE_RX_FLYSKY
+
+#include "build/build_config.h"
+#include "build/debug.h"
+
+#include "common/maths.h"
+#include "common/utils.h"
+#include "config/config_eeprom.h"
+#include "config/parameter_group_ids.h"
+#include "fc/config.h"
+
+#include "rx/rx.h"
+#include "rx/rx_spi.h"
+#include "rx/flysky.h"
+#include "rx/flysky_defs.h"
+
+#include "drivers/rx_a7105.h"
+#include "drivers/system.h"
+#include "drivers/time.h"
+#include "drivers/io.h"
+
+#include "sensors/battery.h"
+
+
+#if FLYSKY_CHANNEL_COUNT > 8

```

Comment:
Is there any advantage for targets if they define channel counts that are less than the maximum for each protocol? If not, then it will simplify things if the channel counts are permanently defined as the max channel counts.

Reply:
Advantage exists. It is necessary to read less bytes over SPI when we define less channel count. This saves time and reduces CPU usage.

Reply:
I see. so the data for the additional channels that are sent by the TX is just discarded in this case?

As long as there is a default, so that target definitions do not *have to* define `FLYSKY_CHANNEL_COUNT` this is fine. But please use something like `MAX_FLYSKY_CHANNEL_COUNT` here and in `flsky_defs.h`.